tools/Google.GCloud.Tools.GenerateSnippetMarkdown/Program.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Google.GCloud.Tools.GenerateSnippetMarkdown
+{
+    /// <summary>
+    /// Simple program to generate a snippets.md file for docfx to consume when generating documentation.
+    /// The file basically links the snippets projects with the client libraries.
+    /// </summary>
+    public sealed class Program
+    {
+        private static readonly Regex StartSnippetTag = new Regex(@"^\s*// \<(?<name>[\w\.]+)>\s*$", RegexOptions.Compiled);
+
+        private static void Main(string[] args)
+        {
+            string root = DetermineRootDirectory();
+            if (root == null)
+            {
+                Console.WriteLine("Unable to determine root directory. Please run within gcloud-dotnet repository.");
+                return;
```

Comment:
Can we use exit codes here? So you can signal error/success for scripting this tool.